src/controller/container-controller.ts

```diff
@@ -140,15 +144,12 @@ export default class ContainerController extends BaseController {
 
     // Handle request
     try {
-      // Check if we should return a 404.
-      const exist = await ContainerRevision.findOne({ where: { container: { id: containerId } } });
-      if (!exist) {
+      const container = (await ContainerService
```

Comment:
Isn't 
`await ContainerRevision.findOne({...ContainerService.getOptions({containerId, returnProducts})})` cleaner? since you force it to be a findOne.

Reply:
But I thought we wanted to be less dependent on different query implementations everywhere? The only thing that `findOne` does compared to `find`, is that `findOne()` basically does `find()[0]`. It is still undefined. By directly calling the service, we only query the entity once (and correctly).

Reply:
I thought it was more about setting "defaults". We can hide defaults within the getOptions (maybe an `archived` property later?)

Reply:
But then what is the difference between using the container service or manually querying a revision with default options? Because the service method does the same thing

Reply:
Not a lot, I just think that its cleaner to "drop" the records[0], imho.

Reply:
(You can resolve this btw, since I am neutral about it given what you said)